### PR TITLE
Adds support for custom date type

### PR DESCRIPTION
**Pull request by alycklama**
I would love to add a test for the custom data type, but it fails to find the specific implicit of `Encoder[org.joda.time.DateTime]`. Which makes sense, I'm just wondering how I can bring it on scope in the test. Do you have any idea perhaps?

### CODE REVIEW

File: core/src/test/scala/io/github/ghostbuster91/sttp/client3/GeneratorTest.scala [hunk 1]
```diff
@@ -124,7 +126,7 @@ object GeneratorTest extends TestSuite {
 
   def testNoCompile(
       codeGenConfig: CodegenConfig = CodegenConfig()
-  )(implicit testPath: utest.framework.TestPath) = {
+  )(implicit testPath: utest.framework.TestPath): Unit = {
```
Comment (alycklama) [inline core/src/test/scala/io/github/ghostbuster91/sttp/client3/GeneratorTest.scala:129]: Gets rid of compiler warnings

File: core/src/test/scala/io/github/ghostbuster91/sttp/client3/GeneratorTest.scala [hunk 2]
```diff
@@ -141,7 +143,7 @@ object GeneratorTest extends TestSuite {
 
   def test(
       codeGenConfig: CodegenConfig = CodegenConfig()
-  )(implicit testPath: utest.framework.TestPath) = {
+  )(implicit testPath: utest.framework.TestPath): Unit = {
```
Comment (alycklama) [inline core/src/test/scala/io/github/ghostbuster91/sttp/client3/GeneratorTest.scala:146]: Gets rid of compiler warnings

**Comment by alycklama** [comment]
Added a test for `Instant` instead of `java.util.Date`, conceptually it doesn't matter.

**Review by ghostbuster91 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**